From 78cf4d71a41265b197e8449719dbdc3f5bbbab5b Mon Sep 17 00:00:00 2001
From: Deepak Rathore <deepak.rathore@einfochips.com>
Date: Wed, 9 Mar 2022 14:31:26 +0530
Subject: [PATCH 09/10] IMX8MQ: Added ARX3A0 camera sensor support

- Added ARX3A0 Camera sensor DTS and added changes in Makefile
  to compile it

Change-Id: I8997707950e6dd05ee63d5f6e88398e77ffdb190
Signed-off-by: Deepak Rathore <deepak.rathore@einfochips.com>
---
 arch/arm64/boot/dts/freescale/Makefile        |  2 +-
 .../freescale/imx8mq-thor96-ap1302-arx3a0.dts | 67 +++++++++++++++++++
 drivers/media/i2c/ap1302.c                    |  4 +-
 3 files changed, 70 insertions(+), 3 deletions(-)
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mq-thor96-ap1302-arx3a0.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index f59fc6099..a272662a1 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -123,7 +123,7 @@ dtb-$(CONFIG_ARCH_MXC) += imx8mq-nitrogen.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mq-phanbell.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mq-pico-pi.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mq-thor96.dtb imx8mq-thor96-dual-display.dtb imx8mq-thor96-dcss-rm67191.dtb \
-			  imx8mq-thor96-lcdif-adv7535.dtb imx8mq-thor96-ap1302-ar0430.dtb
+			  imx8mq-thor96-lcdif-adv7535.dtb imx8mq-thor96-ap1302-ar0430.dtb imx8mq-thor96-ap1302-arx3a0.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mq-evk-dp.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mq-zii-ultra-rmb3.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mq-zii-ultra-zest.dtb
diff --git a/arch/arm64/boot/dts/freescale/imx8mq-thor96-ap1302-arx3a0.dts b/arch/arm64/boot/dts/freescale/imx8mq-thor96-ap1302-arx3a0.dts
new file mode 100644
index 000000000..da4af9a9c
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mq-thor96-ap1302-arx3a0.dts
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright 2019 NXP.
+ */
+
+#include "imx8mq-thor96.dts"
+
+&ov5640_mipi2 {
+	status = "disabled";
+};
+
+&ov5640_mipi {
+	status = "disabled";
+};
+
+&i2c3 {
+	ap1302_mipi: ap1302@3d {
+		compatible = "onnn,ap1302";
+		reg = <0x3d>;
+
+		pinctrl-0 = <&pinctrl_csi>;
+		clocks = <&clk IMX8MQ_CLK_CLKO2>;
+		clock-names = "xclk";
+		assigned-clocks = <&clk IMX8MQ_CLK_CLKO2>,
+				  <&clk IMX8MQ_CLK_CLKO2>;
+		assigned-clock-parents = <&clk IMX8MQ_SYS2_PLL_200M>;
+		assigned-clock-rates = <0>, <20000000>;
+
+		reset-gpios = <&gpio3 24 GPIO_ACTIVE_LOW>;
+		standby-gpios = <&gpio2 7 GPIO_ACTIVE_HIGH>;
+		i2csel-gpios = <&gpio2 8 GPIO_ACTIVE_HIGH>;
+		isptrig-gpios = <&gpio3 12 GPIO_ACTIVE_HIGH>;
+		status = "okay";
+
+		port {
+			ap1302_ep: endpoint {
+				remote-endpoint = <&mipi1_sensor_ep>;
+				data-lanes = <1 2 3 4>;
+			};
+		};
+
+		sensors {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			onnn,model = "onnn,arx3a0";
+			sensor@0 {
+				reg = <0>;
+			};
+		};
+	};
+};
+
+&mipi1_sensor_ep {
+	remote-endpoint = <&ap1302_ep>;
+	data-lanes = <1 2 3 4>;
+	csis-wclk;
+};
+
+&iomuxc {
+	pinctrl_csi: csi_grp {
+		fsl,pins = <
+			MX8MQ_IOMUXC_SD1_DATA5_GPIO2_IO7	0x19
+			MX8MQ_IOMUXC_SAI5_RXD3_GPIO3_IO24	0x19
+			MX8MQ_IOMUXC_GPIO1_IO15_CCMSRCGPCMIX_CLKO2      0x59
+		>;
+	};
+};
diff --git a/drivers/media/i2c/ap1302.c b/drivers/media/i2c/ap1302.c
index ba2e680fe..214d6b9c1 100644
--- a/drivers/media/i2c/ap1302.c
+++ b/drivers/media/i2c/ap1302.c
@@ -542,8 +542,8 @@ static const struct ap1302_sensor_info ap1302_sensor_info[] = {
 		.model = "onnn,arx3a0",
 		.name = "arx3a0",
 		.i2c_addr = 0x36,
-		.resolution = { 1920, 1080 },
-		.fps = 30,
+		.resolution = { 600, 600 },
+		.fps = 120,
 		.format = MEDIA_BUS_FMT_UYVY8_2X8,
 		.supplies = (const struct ap1302_sensor_supply[]) {
 			{ "vaa", 0 },
-- 
2.17.1

